test/samples/async.js

```diff
@@ -138,5 +138,42 @@ module.exports = [
 		options: { transforms: { asyncAwait: true, dangerousForOf: true } },
 		input: `for await (const x of someFunction()) { x() }`,
 		error: /Transforming for-await-of statements is not implemented/
+	},
+
+	// variables interaction with async :
+	{
+		description: 'transpiles a variable decleration with await',
+		options: { transforms: { asyncAwait: true } },
+		input: `async function f() { const a = await f1(); return a + 1; }`,
+		output: `function f() { return Promise.resolve().then(function(){ return f1(); }).then(function(a){ return a + 1;})`
+	},
+
+	{
+		description: 'transpiles multiple variable declerations in a row',
+		options: { transforms: { asyncAwait: true } },
+		input: `async function f() { const a1 = await f1(); const a2 = await f2(); return a1 + a2; }`,
+		output: `function f() { const a1; const a2; return Promise.resolve().then(function(){ return f1(); }).then(function(ref){ a1 = ref; return f2(); }).then(function(ref) { a2 = ref; return a1 + a2; }); }`
```

Comment:
You cannot declare `const` variables without initializing them, so you should use `var` instead.